lib/Model/Purchase/Payment.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fazland\FattureInCloud\Model\Purchase;
+
+class Payment implements \JsonSerializable
+{
+    /**
+     * @var \DateTime
+     */
+    private $expireDate;
+
+    /**
+     * @var string
+     */
+    public $method;
+
+    /**
+     * @var string
+     */
+    public $amount;
+
+    /**
+     * @var \DateTime
+     */
+    private $paymentDate;
+
+    public function __set($name, $value)
+    {
+        switch ($name) {
+            case 'expireDate':
+            case 'paymentDate':
+                $value = str_replace('/', '-', $value);
```

Comment:
As you specified in the above PHPDocs, these two properties should be instances of \DateTime(Immutable)

Reply:
Why do u prefer to use DateTimeImmutable instead DateTime?

Reply:
Because those dates do not change.  Immutable dates should always be preferred from mutable ones, it helps preventing bugs caused by code execution. Just to give you some examples about it: https://medium.com/@codebyjeff/whats-all-this-immutable-date-stuff-anyway-72d4130af8ce

Line 34 should be anyway changed in something like `$value = new \DateTimeImmutable(\str_replace('/', '-', $value);`, otherwise no \DateTime instance will be created